File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    possible = True
    index = [0] * 3
    while index[2] < len(goal):
        if index[0] < len(cards1) and cards1[index[0]] == goal[index[2]]:
            index[0] += 1
            index[2] += 1
            continue
        if index[1] < len(cards2) and cards2[index[1]] == goal[index[2]]:
            index[1] += 1
            index[2] += 1
            continue
        possible = False
        break
    if possible:
        return "Yes"
    else:
        return "No"

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        int[] index = new int[3];
        boolean possible = true;
        while(index[2] < goal.length){
            if(index[0] < cards1.length && cards1[index[0]].equals(goal[index[2]])){
                index[0]++;
                index[2]++;
                continue;
            }
            if(index[1] < cards2.length && cards2[index[1]].equals(goal[index[2]])){
                index[1]++;
                index[2]++;
                continue;
            }
            possible = false;
            break;
        }
        if(possible){
            return "Yes";
        } else {
            return "No";
        }
    }
}
